File: C++/3. Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int ans = 0;
        unordered_map<char, int>maple;
        int last = 0;
        for(int i=0;i<s.length();i++)
        {
            char c = s[i];
            if(maple.find(c) != maple.end())
                last = max(maple[c]+1, last);
            maple[c] = i;
            ans = max(ans, i - last + 1);
        }
        return ans;
    }
};

File: Python/3. Longest Substring Without Repeating Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        ans, last = 0, 0
        maple = dict()
        for i in range(len(s)):
            if maple.get(s[i], -1) != -1:
                last = max(last, maple[s[i]] + 1)
            maple[s[i]] = i
            ans = max(ans, i - last + 1)
        return ans